### ISSUE TITLE
makePrismaClientClass has incorrect TS signature

### ISSUE THREAD

**Issue by superhawk610** [prisma/prisma#3508]
**Describe the bug**
The function argument signature for `prisma-client-lib/makePrismaClientClass` appears to be incorrect. The module declaration is as follows:

```ts
import { Model } from './types';
export declare function makePrismaClientClass<T>({ typeDefs, endpoint, secret, models, }: {
    typeDefs: string;
    endpoint: string;
    secret?: string;
    models: Model[];
}): T;
```

The relevant portion of my generated `prisma-client/index.ts` is:

```ts
/**
 * Type Defs
 */

export const Prisma = makePrismaClientClass<ClientConstructor<Prisma>>({
  typeDefs,
  endpoint: `http://localhost:4466/prisma/dev`
});
export const prisma = new Prisma();
```

The generated error from `tsc` is:

```
src/prisma-client/index.ts:1359:72 - error TS2345: Argument of type '{ typeDefs: string; endpoint: string; }' is not assignable to parameter of type '{ typeDefs: string; endpoint: string; secret?: string; models: Model[]; }'.
  Property 'models' is missing in type '{ typeDefs: string; endpoint: string; }'.
```

Unless I'm missing something, it appears that `models` should also be marked as optional - eg

```ts
models?: Model[]
```

**To Reproduce**
1. Create a new local Prisma deployment using the instructions listed [here](https://www.prisma.io/docs/get-started/01-setting-up-prisma-new-database-TYPESCRIPT-t002/#read-and-write-data-using-the-prisma-client) for the TypeScript default config.
2. `docker-compose up -d`
3. `prisma deploy`
4. `prisma generate`

**Expected behavior**
No errors are thrown by `tsc`.

**Versions (please complete the following information):**
 - OS: Ubuntu 18.10
 - `prisma` CLI: prisma/1.20.1 (linux-x64) node-v11.0.0
 - Prisma Server: prismagraphql/prisma:1.20

**Comment by chanlito** [comment]
@superhawk610 I'm also getting this typescript compilation error too, and many others too. Let's hope somebody takes a look into all these typescript errors because it's important.

**Comment by superhawk610** [comment]
@chanlito What other errors are you seeing? I had 20 or so upon first compile, I had to install `@types/ws` and modify my `tsconfig.json` like so:

```json
{
  "compilerOptions": {
    "skipLibCheck": true
  }
}
```

As a temporary fix, you can add the following file at `@types/prisma-client-lib/makePrismaClientClass.d.ts` to your project:

```ts
import { Model } from 'prisma-client-lib/types';

declare module 'prisma-client-lib' {
  export declare function makePrismaClientClass<T>({
    typeDefs,
    endpoint,
    secret,
    models,
  }: {
    typeDefs: string;
    endpoint: string;
    secret?: string;
    models?: Model[];
  }): T;
}
```

**Comment by chanlito** [comment]
I had to add `models: []` to the generated file to fix this issue.

**Comment by JensMadsen** [comment]
I experience the same error. I wonder why this was not caught in a test?

**Comment by JensMadsen** [comment]
found this one :-) https://github.com/prisma/prisma/issues/3453

**Comment by kevinmarrec** [comment]
It's a regression around `prisma-client-lib`, I think.

I had not the error before upgrading it from `prisma-client-lib@1.20.1` to `prisma-client-lib@1.20.3`.

**Comment by kevinmarrec** [comment]
Can you guys confirm your `prisma-client-lib` version is superior to 1.20.1 ?

**Comment by kevinmarrec** [comment]
Alright, it's indeed a regression.

`yarn remove prisma-client-lib && yarn add prisma-client-lib@1.20.1` fixes the issue.

**Comment by tadejstanic** [comment]
@kevinmarrec I'm using `prisma-client-lib@1.20.3` and I'm pretty sure it's a regression bug.

**Comment by JensMadsen** [comment]
@kevinmarrec i was on 1.20.3 doing yarn remove prisma-client-lib && yarn add prisma-client-lib@1.20.1 as you suggested fixed the issue for me. Thanks :-)

**Comment by kevinmarrec** [comment]
@tadejstanic Yes it's sure
@JensMadsen No problem, Prisma team is working on a patch :)

**Comment by timsuchanek** [comment]
Thanks for reporting @superhawk610 ! We just released `1.20.5` bringing the client and the cli back in sync!
When you're now generating the client with `prisma@1.20.5` and use `prisma-client-lib@1.20.5` it should work as expected.

If that is not the case, please let us know and we reopen this issue!

**Comment by forgo** [comment]
I seem to be having this issue with newer versions:

`prisma@1.20.7` and `prisma-client-lib@1.20.7`

**Comment by xiaoyu-tamu** [comment]
@forgo I was facing the same issue., fixed by install  latest version of `prisma` via npm (previously installed via homebrew)

**Comment by iquabius** [comment]
I was also getting this error on `prisma-client-lib@1.21.0`. I downgraded it to version `1.20.1` for now.

**Comment by lterfloth** [comment]
Had the same problem on my machine but downgrading both versions to 1.20.5 worked for me. The second time the last 2 weeks I had to spend time fixing stuff that should not break in the first place...

**Comment by lterfloth** [comment]
So I'm currently facing an issue that has been fixed in 1.20.7 but I'm unsure whether I can safely update to the newest version yet. Any updates?

**Comment by michaelmafort** [comment]
Keep the prisma and prisma-client-lib in the same version.
```
npm install --save prisma-client-lib
npm install -g prisma
```
Generate new files with correct lib version
```
prisma generate
```
